forge/forge/config/config.py

```diff
@@ -248,7 +248,7 @@ def assert_config_has_openai_api_key(config: Config) -> None:
             print(f"{Fore.RED}Invalid OpenAI API key{Fore.RESET}")
             exit(1)
     # If key is set, but it looks invalid
-    elif not re.search(key_pattern, openai_api_key):
+    elif not config.azure_config_file and not re.search(key_pattern, openai_api_key):
```

Comment:
This check is incomplete because `azure_config_file` is set to a path by default. Instead, you should check whether `openai_credentials.api_type == "azure"`.